internal/firmware/revision.go

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package firmware
+
+import (
+	"fmt"
+	"strconv"
+)
+
+type Revision struct {
+	N int
+}
+
+func (r Revision) String() string {
+	if r.N == 0 {
+		return "unset"
+	}
+	if r.N < 0 {
+		return fmt.Sprintf("x%d", -r.N)
+	}
+	return strconv.Itoa(r.N)
+}
+
+func (r Revision) Unset() bool {
+	return r.N == 0
+}
+
+func (r Revision) Local() bool {
+	return r.N < 0
+}
+
+func (r Revision) Store() bool {
```

Comment:
`Store` is ambiguous, please comment.

Reply:
Hi Harry. You are right this does appear ambiguous, but in the context of store-backed products, Local and Store revisions are well understood concepts. I took this file verbatim from the snapd codebase, and again in that context the idea of a "store revision" is very well understood. I will add comments as you propose.